gcs-fetcher/README.md

```diff
@@ -52,12 +52,79 @@ only writing a new entry for `Dockerfile` to specify the new file's location in
 Cloud Storage.
 
 In the case of a file move, where file contents themselves don't change, no new
-files have to be uploaded, the new manifest simply changes the key in the object
-describing the path to place the file fetched from Cloud Storage.
+files have to be uploaded either. The new manifest simply changes the key in the
+object describing the path to place the file fetched from Cloud Storage.
+
+## Full Example
+
+To fetch source described in a source manifest:
+
+```
+steps:
+- name: 'gcr.io/cloud-builders/gcs-fetcher'
+  args:
+  - '--type=Manifest'
+  - '--location=gs://${PROJECT_ID}_cloudbuild/manifest-foo.json'
+```
+
+
+It may also be useful to _produce_ and upload source manifests describing some
+source, which you can do with `gcr.io/cloud-builders/gcs-uploader`:
+
+```
+steps:
+- name: 'gcr.io/cloud-builders/gcs-uploader'
+  args: ['--bucket=${PROJECT_ID}_cloudbuild']
+```
+
+This will upload the contents of the workspace directory, ignoring objects that
+are already present in Cloud Storage, and upload a randomly named manifest JSON
+object to Cloud Storage. You can specify the location of that manifest with the
+`--manifest_file` flag.
+
+### Caching resources
+
+The two builders can be used together to provide simple cross-build caching
+functionality, by optimistically fetching files that are expensive to generate
+at the beginning of the build, and by uploading those files at the end of the
+build.
+
+In order to benefit from this you would need to define a well-known reusable
+manifest file location.
+
+```
+steps:
+# Attempt to fetch whatever files are available.
+- name: 'gcr.io/cloud-builders/gcs-fetcher'
+  args:
+  - '--type=Manifest'
+  - '--location=gs://${PROJECT_ID}_cloudbuild_cache/manifest-foo.json'
+
+# Generate new files, ignoring those that already exist.
+- name: 'generate-new-files'
+
+# Upload all files; only new files will be uploaded to Cloud Storage.
```

Comment:
...only new and changed files will be uploaded...

Also suggest adding a specific note that files in the bucket that are not in the local workspace will not be deleted.